validation/validation.go

```diff
@@ -0,0 +1,26 @@
+package validation
+
+import (
+	"errors"
+
+	"github.com/preslavmihaylov/todocheck/config"
+	"github.com/preslavmihaylov/todocheck/fetcher"
+	"github.com/preslavmihaylov/todocheck/issuetracker"
+)
+
+// Validate checks if the user has access to the configured repo
+func Validate(issueTracker config.IssueTracker, origin string) error {
+	url, err := issuetracker.HealthcheckURL(issueTracker, origin)
+	if err != nil {
+		if errors.Is(err, issuetracker.ErrUnsupportedHealthCheck) {
+			return nil
+		}
+		return err
+	}
+
+	if !fetcher.IsHealthy(issueTracker, url) {
+		return errors.New("user does not have access to repo")
```

Comment:
Please include the repo URL which is problematic in the error message.

Additionally, this can be changed to:
```
permission denied for repository %s
```

Reply:
No idea how I missed that. Updated!